src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1403,17 +1418,21 @@ invalidateMutableAllocs opts sc cc cs = do
   sub <- use setupValueSub
 
   let mutableAllocs = Map.filter (view isMut) $ cs ^. MS.csPreState . MS.csAllocs
-      allocPtrs =
-        (\(ptr, spec) ->
-           ( ptr
-           , _allocSpecBytes spec
-           , mconcat
-             [ "state of memory allocated in precondition (at "
-             , pack . show . W4.plSourceLoc $ spec ^. allocSpecLoc
-             , ") not described in postcondition"
-             ]
-           )
-        ) <$> Map.elems (Map.intersectionWith (,) sub mutableAllocs)
+      allocPtrs = catMaybes $ map
+        (\case
+          (ptr, spec)
+            | Just sz <- asUnsignedConcreteBv (_allocSpecBytes spec) ->
+              Just
+                ( ptr
+                , fromIntegral sz
+                , mconcat
+                  [ "state of memory allocated in precondition (at "
+                  , pack . show . W4.plSourceLoc $ spec ^. allocSpecLoc
+                  , ") not described in postcondition"
+                  ]
+                )
+            | otherwise -> Nothing)
```

Comment:
Could it ever cause trouble to not have arrays of symbolic size in this list?

Reply:
this is an optimization to not invalidate allocations that are completely overwritten in the postcondition. Checking that the allocation size and the points-to size are equal is more complicated when the sizes are symbolic, so the optimization is effectively disabled for symbolic size allocations. At this point, I don't expect any issues, but it may become necessary to do something later...

Reply:
Okay, great. I agree. We can optimize it later if needed.